Remove ready_fn, self.proc_info

pbaughman opened the pull request:
The ready_fn will be deprecated in the future in favor of the ReadyToTest() action in launch_testing. See ros2/launch#346 (comment) for background information

Signed-off-by: Pete Baughman <pete.baughman@apex.ai>

sloretz commented:
@pbaughman it looks like there's a merge conflict. Would you mind rebasing this one?

pbaughman commented:
@sloretz Yeah, I'll probably need to update some more tests, too.  I think the MR causing the conflict added some new stuff as well

sloretz commented:
CI (Testing `ros2action ros2interface ros2msg ros2node ros2pkg ros2service ros2srv ros2topic`)
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8523)](http://ci.ros2.org/job/ci_linux/8523/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4405)](http://ci.ros2.org/job/ci_linux-aarch64/4405/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=6948)](http://ci.ros2.org/job/ci_osx/6948/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8437)](http://ci.ros2.org/job/ci_windows/8437/)

sloretz commented:
@pbaughman Looks like there are 4 test failures with the same message. Mind taking a look?

```
Exception: generate_test_description functions without a ready_fn argument must return a LaunchDescription containing a ReadyToTest action
```

pbaughman commented:
@sloretz Yeah - sorry - the required changes to launch_testing weren't in the nightly docker image yet, and then I put this down and forgot about it in the middle of adding launch to the workspace.  I'll get it working first thing tomorrow

pbaughman commented:
@sloretz This looks like a bug/not implemented thing in launch.actions.ExecuteProcess.  The ExecuteProcess action isn't returning any sub-entities, so launch_testing can't discover the ReadyToTest action.

I'll need to address that in launch before we can move ahead with this

pbaughman commented:
@sloretz I think that @hidmic fixed the issue we were running into in launch_testing [here](https://github.com/ros2/launch/issues/347)

I noticed that one more test was added that did things the 'old' way.  I've rebased this branch and updated the new test.  Can we try to run this through CI again?

pbaughman commented:
Looks like something is still a little wrong

```
09:23:17 ============================= test session starts ==============================
09:23:17 platform linux -- Python 3.6.8, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
09:23:17 rootdir: /tmp/ws/src/ros2cli/ros2topic, inifile:
09:23:17 plugins: launch-testing-0.9.5, launch-testing-ros-0.9.3, ament-xmllint-0.8.1, ament-pep257-0.8.1, ament-lint-0.8.1, ament-flake8-0.8.1, ament-copyright-0.8.1, cov-2.5.1
09:23:17 collected 8 items
09:23:17 
09:24:02 test/test_cli.py .                                                       [ 12%]
09:24:02 test/test_copyright.py .                                                 [ 25%]
09:24:17 test/test_echo_pub.py F                                                  [ 37%]
09:24:18 test/test_flake8.py .                                                    [ 50%]
09:24:19 test/test_info.py .                                                      [ 62%]
09:24:19 test/test_pep257.py .                                                    [ 75%]
09:24:19 test/test_qos_conversions.py .                                           [ 87%]
09:24:19 test/test_xmllint.py .                                                   [100%]
09:24:19 
09:24:19 -------- generated xml file: /tmp/ws/test_results/ros2topic/pytest.xml ---------
09:24:19 =================================== FAILURES ===================================
09:24:19 _________________________ launch tests: test_echo_pub __________________________
09:24:19 some test cases have failed:
09:24:19 
09:24:19 ----------------------------- Captured stdout call -----------------------------
09:24:19 [INFO] [launch]: All log files can be found below /home/buildfarm/.ros/log/2019-11-18-09-23-17-682477-f87f49776edc-3095
09:24:19 [INFO] [launch]: Default logging verbosity is set to INFO
09:24:19 [INFO] [daemon-stop-35]: process started with pid [3299]
09:24:19 [INFO] [daemon-stop-35]: process has finished cleanly [pid 3299]
09:24:19 [INFO] [daemon-start-36]: process started with pid [3302]
09:24:19 [INFO] [daemon-start-36]: process has finished cleanly [pid 3302]
09:24:19 Timed out waiting for processes to start up
09:24:19 Processes under test stopped before tests completed
09:24:19 ===================== 1 failed, 7 passed in 61.80 seconds ======================
09:24:19 
```

pbaughman commented:
@sloretz I think this is good to go after rebasing.  Can you run CI?  I don't get any ros2topic failures when I run the tests locally on 'master' and on my branch.

I **do** see failures in ros2actions but those happen for me in 'master'

pbaughman commented:
@sloretz Can you re-run CI on this?  I don't see any failure in this branch that aren't already in 'master'

pbaughman commented:
@sloretz Sorry to keep bugging you, but *poke*  ^^

sloretz commented:
CI (Testing ros2action ros2interface ros2msg ros2node ros2pkg ros2service ros2srv ros2topic)
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9262)](http://ci.ros2.org/job/ci_linux/9262/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4948)](http://ci.ros2.org/job/ci_linux-aarch64/4948/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7572)](http://ci.ros2.org/job/ci_osx/7572/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=9183)](http://ci.ros2.org/job/ci_windows/9183/)
* Windows-container [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows-container&build=57)](http://ci.ros2.org/job/ci_windows-container/57/)

pbaughman commented:
@sloretz Looks like something with the build failed?

```
11:03:07 Package 'ros2msg' specified with --packages-up-to was not found
11:03:07 Package 'ros2srv' specified with --packages-up-to was not found
```

It doesn't look like it even made it to the 'test' step  I'm guessing these packages were renamed or something on 'master'

I rebased the branch again

sloretz commented:
> It doesn't look like it even made it to the 'test' step I'm guessing these packages were renamed or something on 'master'

Ah oops. I copy pasted the args I used last time, but it looks like those were deleted in #407.

CI (Testing ros2action ros2interface ros2node ros2pkg ros2service ros2topic)
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9264)](http://ci.ros2.org/job/ci_linux/9264/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4949)](http://ci.ros2.org/job/ci_linux-aarch64/4949/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7572)](http://ci.ros2.org/job/ci_osx/7572/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=9184)](http://ci.ros2.org/job/ci_windows/9184/)
* Windows-container [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows-container&build=58)](http://ci.ros2.org/job/ci_windows-container/58/)

pbaughman commented:
@sloretz I'm pretty sure this is a bad test that tries to assert about the `fibonacci_action_server.py` process before it's started up.  Should I open a different PR that attempts to fix the test, or should I try to glom it onto this branch?

@hidmic Just FYI if you're seeing similar problems on other similar tests ^^

pbaughman commented:
I'll need to fix https://github.com/ros2/launch/issues/381 before I can fix this test.  I want to add a stdout_ready_listener to delay the start of the test until the fibonacci_action_server.py process is started up, but I can't because event_handler introspection appears to be broken right now

pbaughman commented:
Waiting on https://github.com/ros2/launch/pull/386 so I can delay the start of the tests until after the Fibonacci action server is up-and-running

pbaughman commented:
@sloretz Ok, https://github.com/ros2/ros2cli/pull/459 should hopefully fix the test that was causing us problems.  Can we try CI again (for the billionth time, I'm so sorry 😢 )

pbaughman commented:
@sloretz ping^^

sloretz commented:
CI (Testing ros2action ros2interface ros2node ros2pkg ros2service ros2topic) :crossed_fingers: 
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9538)](http://ci.ros2.org/job/ci_linux/9538/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=5245)](http://ci.ros2.org/job/ci_linux-aarch64/5245/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7791)](http://ci.ros2.org/job/ci_osx/7791/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=9462)](http://ci.ros2.org/job/ci_windows/9462/)

pbaughman commented:
This looks unrelated to my changes again. . . 
```
17:55:30     from rmw_implementation import get_available_rmw_implementations
17:55:30 E   ModuleNotFoundError: No module named 'rmw_implementation'
```

I'll try rebasing again I guess.

pbaughman commented:
@sloretz Yeah, 96cf5b94 changed this yesterday and I pushed to this branch five days ago.  Please run it again I guess.  I'm going to be so happy when this PR is gone forever.

sloretz commented:
CI (Testing ros2action ros2interface ros2node ros2pkg ros2service ros2topic)
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9539)](http://ci.ros2.org/job/ci_linux/9539/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=5246)](http://ci.ros2.org/job/ci_linux-aarch64/5246/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7792)](http://ci.ros2.org/job/ci_osx/7792/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=9463)](http://ci.ros2.org/job/ci_windows/9463/)


> I'm going to be so happy when this PR is gone forever.

Thank you for sticking with it!

> Should I just log in every day and rebase this branch while I'm waiting for someone to run CI? This is not the first time that it's gotten broken out from under me. How can I make faster progress here?

Rebasing frequently helps. That works around ci.ros2.org being incapable of merging forks into the target branch and then building/testing.

sloretz commented:
Gah, Wrong parameters CI again

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9544)](http://ci.ros2.org/job/ci_linux/9544/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=5250)](http://ci.ros2.org/job/ci_linux-aarch64/5250/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7796)](http://ci.ros2.org/job/ci_osx/7796/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=9467)](http://ci.ros2.org/job/ci_windows/9467/)

pbaughman commented:
@sloretz Still looks like upstream issues with the build.  It didn't even make it to the tests:

```
--- output: sqlite3_vendor
14:27:15 [0.045s] CMake Error: The source directory "/home/jenkins-agent/workspace/ci_linux/ws/build/sqlite3_vendor/rosbag2" does not exist.
14:27:15 [0.045s] Specify --help for usage, or press the help button on the CMake GUI.
14:27:15 [0.045s] Not searching for unused variables given on the command line.
14:27:15 ---
14:27:15 Aborted  <<< sqlite3_vendor
```

No idea what's been going on with sqllite3_vendor or rosbag2.  Gonna rebase again?

jacobperron commented:
The CI jobs linked do not look like the correct ones.

sloretz commented:
> The CI jobs linked do not look like the correct ones.

Good catch. It looks like the jobs were queue'd, but some earlier in the queue got aborted so the real job numbers didn't match. I've fixed the links in the comment above. @pbaughman FYI

jacobperron commented:
The failures using CycloneDDS is a known issue: https://github.com/ros2/ros2cli/issues/457

jacobperron commented:
Connext failures also appear, sometimes, on the nightlies ([for example](https://ci.ros2.org/view/nightly/job/nightly_linux_repeated/1790/testReport/ros2topic.test/test_cli/test_cli/)).

pbaughman commented:
> The failures using CycloneDDS is a known issue: #457

It looks like that's all the failures in ros2service.test.test_cli.test_cli, ros2node.test.test_cli.test_cli, and ros2action.test.test_cli.test_cli were cyclone DDS failures.

The connext failures seem like they're only in ros2topic.test.test_cli.test_cli

jacobperron commented:
@sloretz I'll defer to you for merging this.

sloretz commented:
Thanks for the PR and for sticking with it @pbaughman !
